core/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -168,7 +174,9 @@
         this.queryResult = new QuerySearchResult(id, shardTarget);
         this.fetchResult = new FetchSearchResult(id, shardTarget);
         this.indexShard = indexShard;
+        this.searchExecutor = indexShard.getThreadPool().executor(ThreadPool.Names.SEARCH);
```

Comment:
I don't think I understand, I need a way from `QueryPhase.execute` to access the executor to get the stats out of it, I can't do it on the consumer end since that may be on a different node, and the only thing available in the (static) `QueryPhase.execute` method is the `SearchContext`.

Is there a better place to put this?